### ISSUE TITLE
sum: should work with empty collection

### ISSUE THREAD

**Issue by Ducasse** [guillep/pharo-core#6373]
#() sum: #linesOfCode should return 0.

right now it forces all the clients to check if the receiver is empty. 
If we do not fix it, we should remove it and the user uses inject:into:

**Comment by Ducasse** [comment]
So just adding a self isEmpty ifTrue: [^ 0 ] would make it a lot more usable

**Comment by svenvc** [comment]
Here we go again ...

(And it totally agree: https://en.wikipedia.org/wiki/Empty_sum)

<blockquote><img src="/static/favicon/wikipedia.ico" width="48" align="right"><div><strong><a href="https://en.wikipedia.org/wiki/Empty_sum">Empty sum - Wikipedia</a></strong></div></blockquote>

**Comment by Ducasse** [comment]
thanks!
And we will fix it or remove it. 
We can add sumNumbers: to help :)

**Comment by VincentBlondeau** [comment]
What about adding sum:ifEmpty:?

`#() sum: #linesOfCode ifEmpty: [0]`

**Comment by bencoman** [comment]
I find that Wikipedia article hard to follow.  I think this is a simpler explanation in favour...
https://math.stackexchange.com/a/2310799

<blockquote><img src="https://cdn.sstatic.net/Sites/math/Img/apple-touch-icon@2.png?v=4ec1df2e49b1" width="48" align="right"><div>Mathematics Stack Exchange</div><div><strong><a href="https://math.stackexchange.com/questions/2310758/empty-set-and-empty-sum">Empty set and empty sum</a></strong></div><div>Empty set is closed under addition. If not, there would be at least one element in the empty set whose addition with itself is not there, contradiction. Thus the result follows. This not a conventi...</div></blockquote>

**Comment by Ducasse** [comment]
Yes vincent I like your idea!
I think that it solves many problem.

**Comment by olekscode** [comment]
This issue should be closed because the PR https://github.com/pharo-project/pharo/pull/6453 has been merged and the duplicated issue #6452 has been closed
